app/views/admin/pages/templates/new.html.erb

```diff
@@ -1,19 +1,20 @@
-<% @page_title = 'Add new page template' %>
+<% @page_title = 'Add new template' %>
 
-<%= form_for :page_template, url: admin_pages_template_add_path, method: 'post' do |f| %>
+<%= form_for :page_template, url: admin_pages_template_new_path, method: :post do |f| %>
   <p>
-    Name:
+    <%= f.label :name %>
     <br><%= f.text_field :name %>
   </p>
   <p>
-    Description:
+    <%= f.label :description %>
     <br><%= f.text_field :description %>
   </p>
   <p>
-    Template filename:
-    <br><%= f.text_field :filename %> <%# TODO: replace with select drop-down %>
+    <%# TODO: replace with select drop-down %>
```

Comment:
Agreed 👍 . Also the submitted value will want to be strictly validated against a safelist of known filenames when saved.

This is to mitigate file traversal attack shenanigans, say if a malicious user tried to submit a filename like `../../../top/secret.txt`. And this is later paired with a separate vulnerability to reveal the contents of that file.

Reply:
How's this? https://github.com/denny/ShinyCMS-ruby/commit/597e8c93663af4

Reply:
Looks solid 👍